lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +193,208 @@ const steps = {
+
+    const orgName = configuration.org;
+    if (orgName) {
+      let providers;
+      try {
+        providers = await getProviders(orgName);
+      } catch (err) {
+        if (err.code === 'DASHBOARD_UNAVAILABLE') {
+          process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+          return false;
+        }
+        throw err;
+      }
+      const hasDefaultProvider = providers.some((provider) => provider.isDefault);
+
+      if (hasDefaultProvider) return false;
+    }
+
+    return true;
   },
   async run(data) {
-    process.stdout.write(
-      'No AWS credentials were found on your computer, ' +
-        'you need these to host your application.\n\n'
-    );
-
     return module.exports.runSteps(data);
   },
   steps,
   runSteps: async (context) => {
-    if (!(await steps.shouldSetupAwsCredentials())) return;
-    await steps.ensureAwsAccount();
+    let providers;
+
+    // It is possible that user decides to not configure org for his service and
+    // we still should allow setup of local credentials in such case
+    if (context.configuration.org) {
+      try {
+        providers = await getProviders(context.configuration.org);
+      } catch (err) {
+        if (err.code === 'DASHBOARD_UNAVAILABLE') {
+          process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+          return;
+        }
+        throw err;
+      }
+    }
+    const credentialsSetupChoiceAnswer = await credentialsSetupChoice(providers);
+
+    if (credentialsSetupChoiceAnswer === CREDENTIALS_SETUP_CHOICE.CREATE_PROVIDER) {
+      try {
+        const createdProviderUid = await steps.handleProviderCreation(context);
+        const hadExistingProviders = Boolean(providers.length);
+        if (createdProviderUid && hadExistingProviders) {
+          // This is situation where user decided to create a new provider and already had previous providers setup
+          // In this case, we want to setup an explicit link between provider and service as the newly created provider
+          // might not be the default one.
+          await steps.linkProviderToService(
+            context.configuration.app,
+            context.configuration.service,
+            context.configuration.org,
+            createdProviderUid
+          );
+        }
+        return;
+      } catch (err) {
+        if (err.code === 'DASHBOARD_UNAVAILABLE') {
+          process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+          return;
+        }
+        throw err;
+      }
+    } else if (credentialsSetupChoiceAnswer === CREDENTIALS_SETUP_CHOICE.SKIP) {
+      steps.writeOnSetupSkip();
+      return;
+    } else if (credentialsSetupChoiceAnswer === CREDENTIALS_SETUP_CHOICE.LOCAL) {
+      await steps.ensureAwsAccount();
+      await steps.ensureAwsCredentials(context);
+      await steps.inputAwsCredentials();
+      return;
+    }
+
+    // Otherwise user selected an existing provider
+    await steps.linkProviderToService(
```

Comment:
Shouldn't we show success message afterwards?

Reply:
great idea :+1: